tests/multi_obj_pool_test.cpp

```diff
@@ -36,29 +36,56 @@ www.navitia.io
 #include <set>
 #include <boost/range/algorithm/find.hpp>
 
+template <typename Obj>
+struct CountParetoFrontVisitor {
+    size_t nb_dominated = 0;
+    size_t nb_dominates = 0;
+    size_t nb_inserted = 0;
+
+    void at_dominated(const Obj&, const Obj&) { ++nb_dominated; }
+    void at_dominates(const Obj&, const Obj&) { ++nb_dominates; }
+    void at_inserted(const Obj&) { ++nb_inserted; }
+};
 
 BOOST_AUTO_TEST_CASE(one_obj) {
     std::function<bool(int, int)> f = [](int a, int b) { return a >= b; };
 
-    ParetoFront<int, std::function<bool(int, int)>> pool(f);
+    ParetoFront<int, std::function<bool(int, int)>, CountParetoFrontVisitor<int>> pool(f);
+    const auto& visitor = pool.get_visitor();
 
     BOOST_CHECK_EQUAL(pool.size(), 0);
 
     BOOST_CHECK(pool.add(1));
     BOOST_REQUIRE_EQUAL(pool.size(), 1);
     BOOST_CHECK_EQUAL(*pool.begin(), 1);
+    BOOST_CHECK_EQUAL(visitor.nb_inserted, 1);
+    BOOST_CHECK_EQUAL(visitor.nb_dominates, 0);
+    BOOST_CHECK_EQUAL(visitor.nb_dominated, 0);
 
     BOOST_CHECK(pool.add(2));
     BOOST_REQUIRE_EQUAL(pool.size(), 1);
     BOOST_CHECK_EQUAL(*pool.begin(), 2);
+    BOOST_CHECK_EQUAL(visitor.nb_inserted, 2);
+    BOOST_CHECK_EQUAL(visitor.nb_dominates, 1);
+    BOOST_CHECK_EQUAL(visitor.nb_dominated, 0);
 
     BOOST_CHECK(! pool.add(1));
     BOOST_REQUIRE_EQUAL(pool.size(), 1);
     BOOST_CHECK_EQUAL(*pool.begin(), 2);
+    BOOST_CHECK_EQUAL(visitor.nb_inserted, 2);
+    BOOST_CHECK_EQUAL(visitor.nb_dominates, 1);
+    BOOST_CHECK_EQUAL(visitor.nb_dominated, 1);
 
     BOOST_CHECK(! pool.add(2));
     BOOST_REQUIRE_EQUAL(pool.size(), 1);
     BOOST_CHECK_EQUAL(*pool.begin(), 2);
+    BOOST_CHECK_EQUAL(visitor.nb_inserted, 2);
+    BOOST_CHECK_EQUAL(visitor.nb_dominates, 1);
+    BOOST_CHECK_EQUAL(visitor.nb_dominated, 2);
+
+    BOOST_CHECK(pool.contains_better_than(1));
+    BOOST_CHECK(pool.contains_better_than(2));
+    BOOST_CHECK(!pool.contains_better_than(3));
```

Comment:
`contains_better_than` can be tested with a more complex object to test the weakly dominated case (worst on one dimension, worst on the others).